### PR TITLE
add logging to decoupleApps

### LINKED ISSUE #1206: [branch: decoupleApps] Add diagnostic logs for application download + extract via decoupling
Add diagnostic logs for application download + extract via decoupling

### PR DESCRIPTION
**Pull request by AlexanderGetka-cbica**
Fixes #1206 

Adds logging to ApplicationDownloadManager class.
Let me know if there are other events I should log, but as it is currently constructed I'm not sure how much more fine-grained we can make these logs.
I only changed ApplicationDownloadManager.h and ApplicationDownloadManager.cpp -- no need to review/worry about the other files. Those are just from the other commits that got auto-included.
EDIT: Also now includes a change to the azure-pipelines.yml file for the branch. This should help clear up the duplicate PR build triggers on Azure.

### CODE REVIEW

**Comment by AlexanderGetka-cbica** [comment]
Seems like Azure is having problems with the Windows VMs they host. The required checks have passed, however.

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@ SET( ${PROJECT_NAME}_Variant "Full" ) # the particular variant of CaPTk (Full/Ne
 
 SET( PROJECT_VERSION_MAJOR 1 )
 SET( PROJECT_VERSION_MINOR 8 )
-SET( PROJECT_VERSION_PATCH 0.nonRelease )
+SET( PROJECT_VERSION_PATCH 0.Beta )
```
Comment (sarthakpati) [inline CMakeLists.txt:16]: Could you change this to `nonRelease` as we are not tagging this?

Comment (AlexanderGetka-cbica) [reply CMakeLists.txt]: Done.

**Comment by ashishsingh18** [comment]
@AlexanderGetka-cbica One of the required Windows builds is failing..

**Comment by AlexanderGetka-cbica** [comment]
Seems like it was a transient error. If it pops up again then that windows machine will need some manual intervention to remove the extra "bin" dir that was causing the issue.

**Comment by AlexanderGetka-cbica** [comment]
Also, this is ready for review

**Pull request merged**